panic if decoding to an empty slice

Try the following code:

```
package main

import "github.com/BurntSushi/toml"
import "log

type Q struct {
	Test []string
}

func main() {
	q := Q{Test: []string{}}
	_, err := toml.Decode(`Test = ["test"]`, &q)
	if err != nil {
		log.Fatal(err)
	}
}
```

This results in a panic:

```
panic: reflect: slice index out of range

goroutine 1 [running]:
reflect.Value.Index(0x6c1b60, 0x12658d80, 0xd7, 0x0, 0x0, 0x0, 0x0)
        c:/go/src/reflect/value.go:818 +0x10f github.com/BurntSushi/toml.(*MetaData).unifySliceArray(0x12652ba0, 0x6c0d60, 0x12658dd0, 0x57, 0x6c1b60, 0x12658d80, 0xd7, 0x0, 0x0)
        %GOPATH%/src/github.com/BurntSushi/toml/decode.go:321 +0xde github.com/BurntSushi/toml.(*MetaData).unifySlice(0x12652ba0, 0x6c0d60, 0x12658dd0, 0x6c1b60, 0x12658d80, 0xd7, 0x0, 0x0)
        %GOPATH%/src/github.com/BurntSushi/toml/decode.go:314 +0x1b4
github.com/BurntSushi/toml.(*MetaData).unify(0x12652ba0, 0x6c0d60, 0x12658dd0, 0x6c1b60, 0x12658d80, 0xd7, 0x0, 0x0)
        %GOPATH%/src/github.com/BurntSushi/toml/decode.go:206 +0xb11 github.com/BurntSushi/toml.(*MetaData).unifyStruct(0x12652ba0, 0x6cc160, 0x126543c0, 0x6d3e80, 0x12658d80, 0xd9, 0x0, 0x0)
        %GOPATH%/src/github.com/BurntSushi/toml/decode.go:252 +0x65a github.com/BurntSushi/toml.(*MetaData).unify(0x12652ba0, 0x6cc160, 0x126543c0, 0
x6d3e80, 0x12658d80, 0xd9, 0x0, 0x0)
        %GOPATH%/src/github.com/BurntSushi/toml/decode.go:200 +0xb9f github.com/BurntSushi/toml.Decode(0x7adf28, 0xf, 0x6bcbe0, 0x12658d80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        %GOPATH%/src/github.com/BurntSushi/toml/decode.go:114 +0x1c5 main.NewStaticContext(0x12655947, 0xa, 0x12656900, 0x0, 0x0, 0x0)
        ...
```

This is only caused when the `Test` field is initialized to an empty slice.

If `Test` is `nil` or has at least one element, `Decode` works as expected.

--- a/decode.go
+++ b/decode.go
@@ -308,7 +308,7 @@ func (md *MetaData) unifySlice(data interface{}, rv reflect.Value) error {
 		return badtype("slice", data)
 	}
 	sliceLen := datav.Len()
-	if rv.IsNil() {
+	if rv.IsNil() || rv.Len() != datav.Len() {
 		rv.Set(reflect.MakeSlice(rv.Type(), sliceLen, sliceLen))
 	}
 	return md.unifySliceArray(datav, rv)

--- a/decode_test.go
+++ b/decode_test.go
@@ -468,6 +468,15 @@ type ingredient struct {
 	Name string
 }
 
+func TestDecodeToEmptySlice(t *testing.T) {
+	s := struct{ Test []string }{Test: []string{}}
+
+	_, err := Decode(`Test = ["test"]`, &s)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
 func ExampleMetaData_PrimitiveDecode() {
 	var md MetaData
 	var err error

